src/content/en/updates/2018/08/embind.md

```diff
@@ -0,0 +1,256 @@
+or C++ code, each file is compiled separately. Then a linker takes care of
+munging all these so-called object files together and turning them into a wasm
+file. With C, the names of the functions are still available in the object file
+for the linker to use. All you need to be able to call a C function is the name,
+which we are providing as a string to `cwrap()`.
+
+C++ on the other hand supports function overloading, meaning you can implement
+the same function multiple times as long as the signature is different (e.g.
+differently typed parameters). At the compiler level, a nice name like `add`
+would get _mangled_ into something that encodes the signature in the function
+name for the linker. As a result, we wouldn't be able to look up our function
+with its name anymore.
+
+Note: You can prevent the compiler from mangling your functions' names by
+ annotating it with `extern "C"`.
+
+## Enter embind
+
+[embind](https://kripken.github.io/emscripten-site/docs/porting/connecting_cpp_and_javascript/embind.html)
+is part of the Emscripten toolchain and provides you with a bunch of C++ macros
+that allow you to annotate C++ code. You can declare which functions, enums,
+classes or value types you are planning to use from JavaScript. Let's start
+simple with some plain functions:
+
+    #include <emscripten/bind.h>
+
+    using namespace emscripten;
+
+    double add(double a, double b) {
+      return a + b;
+    }
+
+    std::string exclaim(std::string message) {
+      return message + "!";
+    }
+
+    EMSCRIPTEN_BINDINGS(my_module) {
+      function("add", &add);
+      function("exclaim", &exclaim)
+    }
+
+Compared to my previous article, we are not including `emscripten.h` anymore, as
+we don't have to annotate our functions with `EMSCRIPTEN_KEEPALIVE` anymore.
+Instead we have a `EMSCRIPTEN_BINDINGS` section in which we list the names under
+which we want to expose our functions to JavaScript.
+
+Note: The parameter for the `EMSCRIPTEN_BINDINGS` macro is mostly used to avoid
+name conflicts in bigger projects.
+
+To compile this file, we can use the same setup (or, if you want, the same
+Docker image) as in the [previous
+article](/web/updates/2018/03/emscripting-a-c-library). To make use of embind,
+we just have to make sure we use the `--bind` flag:
+
+    $ emcc --bind -O3 add.cpp
+
+Now all that's left to do is whipping up an HTML file that loads our freshly
+created wasm module:
+
+    <script src="/a.out.js"></script>
+    <script>
+    Module.onRuntimeInitialized = _ => {
+      console.log(Module.add(1, 2.3));
+      console.log(Module.exclaim("hello world"));
+    };
+    </script>
+
+Note: If you are curious, I wrote up the same C++ module _without_ embind to
+give you a feel for how much work it is doing for you. [Take a look at the
+gist](https://gist.github.com/surma/d04cd0fd896610575126d30de36d7eb6).
+
+As you can see, we aren't using `cwrap()` anymore — this just works straight out
```

Comment:
My eyes keep reading that as 'crap' and I keep doing a double take.

Oh, and 'This just works' is the start of a new sentence.